### PR TITLE
#137 - Added Project Lombok lib

### PR DESCRIPTION
**Pull request by aschworer**
#137 - Added Project Lombok lib in pom.xml
Added @Getter, @Setter, @AllArgsConstructor annotations.
Removed all getters and setters, instead they will be generated automatically during the build.

### CODE REVIEW

**Comment by aschworer** [comment]
There shouldn't be any issues with this change, but if there are, I'll see why and either revert the change, or fix it.

File: src/main/java/ai/elimu/model/enums/analytics/TaskType.java [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import java.util.Set;
 import ai.elimu.model.enums.content.LiteracySkill;
 import ai.elimu.model.enums.content.NumeracySkill;
+import lombok.AllArgsConstructor;
 
 // TODO: rename to "LearningType" or "ActivityType"?
+@AllArgsConstructor
 public enum TaskType {
```
Comment (jo-elimuai) [inline src/main/java/ai/elimu/model/enums/analytics/TaskType.java]: I'm getting an error message in the editor (NetBeans): 

screen shot 2017-11-14 at 00 03 10

File: src/main/java/ai/elimu/model/gson/content/multimedia/AudioGson.java [hunk 2]
```diff
@@ -1,26 +1,17 @@
...
+    @Getter
+    @Setter
+    private String transcription;
 
-    public AudioFormat getAudioFormat() {
-        return audioFormat;
-    }
+    @Getter
+    @Setter
+    private AudioFormat audioFormat;
```
Comment (jo-elimuai) [inline src/main/java/ai/elimu/model/gson/content/multimedia/AudioGson.java]: The webapp no longer compiles:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/jogrimst/GitHub/webapp/src/main/java/ai/elimu/rest/v1/JavaToGsonConverter.java:[80,22] cannot find symbol
  symbol:   method setAudioType(ai.elimu.model.enums.content.AudioFormat)
  location: variable audioGson of type ai.elimu.model.gson.content.multimedia.AudioGson
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
```

Comment (aschworer) [reply src/main/java/ai/elimu/model/gson/content/multimedia/AudioGson.java]: Weird, I thought did build webapp successfully before pushing. I'll have a look

Comment (aschworer) [reply src/main/java/ai/elimu/model/gson/content/multimedia/AudioGson.java]: my bad, my built was using old model.
pushed the fix for this. plz check if its ok now

File: src/main/java/ai/elimu/model/enums/analytics/TaskType.java (hunk 1 above)
Comment (aschworer) [reply src/main/java/ai/elimu/model/enums/analytics/TaskType.java]: I guess you need to tune your NetBeans to recognise Lombok. Try this? https://projectlombok.org/setup/netbeans

Comment (aschworer) [reply src/main/java/ai/elimu/model/enums/analytics/TaskType.java]: Is it good now? Did you try this plugin for NetBeans?

**Pull request closed without merging**